Spatial.gd

```diff
@@ -490,101 +498,89 @@ func init_category_tree():
 
 
 func waypoint_categories_to_godot_nodes():
-	for category in self.waypoint_data.get_category():
-		_waypoint_categories_to_godot_nodes(null, category, category.get_name(), false)
+	for waypoint_category in self.waypoint_data.get_category():
+		_waypoint_categories_to_godot_nodes(null, waypoint_category, self.categories, self.minimap, false)
 
 
-func _waypoint_categories_to_godot_nodes(item: TreeItem, category, full_category_name: String, collapsed: bool):
-	var category_item: TreeItem = self.marker_packs.create_item(item)
-	if category.get_name() == "":
-		# If this is called, there is an error in the Waypoint data
-		category_item.set_text(0, "No name")
-		category_item.set_metadata(0, "")
+func _waypoint_categories_to_godot_nodes(item: TreeItem, waypoint_category, parent_category: Spatial, parent_category2d: Node2D, collapsed: bool):
+	if waypoint_category.get_name() == "":
+		# If this is called, there is an empty Category in the Waypoint data
```

Comment:
Though testing, I found that the case of a waypoint_category being present but empty is a bug because protobuf does not remove messages that have had the content removed. This should be addressed in the xml_converter. I'll open an issue for it.

Reply:
It seems this bug was already fixed and is no longer an issue. I still think for UI reasons, I think something should be added as a placeholder.